dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEnrollmentAnalyticsTableManager.java

```diff
@@ -0,0 +1,345 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import static org.hisp.dhis.commons.util.TextUtils.removeLast;
+import static org.hisp.dhis.system.util.MathUtils.NUMERIC_LENIENT_REGEXP;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+
+import org.hisp.dhis.analytics.AnalyticsTable;
+import org.hisp.dhis.analytics.AnalyticsTableColumn;
+import org.hisp.dhis.common.ValueType;
+import org.hisp.dhis.commons.collection.UniqueArrayList;
+import org.hisp.dhis.dataelement.DataElement;
+import org.hisp.dhis.organisationunit.OrganisationUnitGroupSet;
+import org.hisp.dhis.organisationunit.OrganisationUnitLevel;
+import org.hisp.dhis.period.PeriodType;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.program.ProgramStage;
+import org.hisp.dhis.program.ProgramStageDataElement;
+import org.hisp.dhis.system.util.DateUtils;
+import org.hisp.dhis.trackedentity.TrackedEntityAttribute;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+
+public class JdbcEnrollmentAnalyticsTableManager
+    extends AbstractEventJdbcTableManager
+{
+    private static final Set<ValueType> NO_INDEX_VAL_TYPES = ImmutableSet.of( ValueType.TEXT, ValueType.LONG_TEXT );
+    public static final String SEP = "_";
+    
+    @Override
+    @Transactional
+    public List<AnalyticsTable> getTables( Date earliest )
+    {
+        return getTables();
+    }
+    
+    @Override
+    public Set<String> getExistingDatabaseTables()
+    {
+        // TODO Auto-generated method stub
+        return null;
+    }
+    
+    private List<AnalyticsTable> getTables() {
+        List<AnalyticsTable> tables = new UniqueArrayList<>();
+        List<Program> programs = idObjectManager.getAllNoAcl( Program.class );
+
+        String baseName = getTableName();
+        
+        for ( Program program : programs )
+        {
+            AnalyticsTable table = new AnalyticsTable( baseName, null, null, program );
+            List<AnalyticsTableColumn> dimensionColumns = getDimensionColumns( table );
+            table.setDimensionColumns( dimensionColumns );
+            tables.add( table );
+        }
+        
+        return tables;
+    }
+    
+    @Override
+    public String getTableName()
+    {
+        return ENROLLMENT_ANALYTICS_TABLE_NAME;
+    }
+
+    @Override
+    public void createTable( AnalyticsTable table )
```

Comment:
Move up to abstract class. Make getDimensionColumns abstract and call from super.